subsystems/shootersubsystem.py

```diff
@@ -0,0 +1,153 @@
+from commands2 import Subsystem
+from wpilib import SmartDashboard
+from wpimath.geometry import Rotation2d
+from util.simtalon import Talon
+from util.simneo import NEOBrushless
+from util.simcoder import CTREEncoder
+import constants
+
+
+class ShooterSubsystem(Subsystem):
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)
+
+        self.angleMotor = Talon(
+            constants.kAngleMotorCANId,
+            constants.kAngleMotorName,
+            constants.kAngleMotorPGain,
+            constants.kAngleMotorIGain,
+            constants.kAngleMotorDGain,
+            constants.kAngleMotorInverted,
+        )
+        self.leftShootingMotor = NEOBrushless(
+            constants.kLeftShootingMotorCANId,
+            constants.kLeftShootingMotorName,
+            constants.kLeftShootingMotorPIDSlot,
+            constants.kLeftShootingMotorPGain,
+            constants.kLeftShootingMotorIGain,
+            constants.kLeftShootingMotorDGain,
+            constants.kLeftShootingMotorInverted,
+        )
+        self.rightShootingMotor = NEOBrushless(
+            constants.kRightShootingMotorCANId,
+            constants.kRightShootingMotorName,
+            constants.kRightShootingMotorPIDSlot,
+            constants.kRightShootingMotorPGain,
+            constants.kRightShootingMotorIGain,
+            constants.kRightShootingMotorDGain,
+            constants.kRightShootingMotorInverted,
+        )
+
+        self.shooterEncoder = CTREEncoder(
+            constants.kShooterAngleEncoderCANId,
+            constants.kShooterAngleMotorOffset.degrees()
+            / constants.kDegeersPerRevolution,
+        )
+
+        self.leftShootingMotor.setSmartCurrentLimit(
+            constants.kShootingMotorCurrentLimit
+        )
+        self.rightShootingMotor.setSmartCurrentLimit(
+            constants.kShootingMotorCurrentLimit
+        )
+
+        # set motor position
+
+        self.angleMotor.setEncoderPosition(
+            constants.kShooterAngleMotorOffset.degrees()
+            / constants.kDegeersPerRevolution
+            * constants.kShootingMotorRatio
+        )
+
+        self.targetAngle = Rotation2d()
+        self.leftTargetSpeed = 0
+        self.rightTargetSpeed = 0
+
+        # speeds are in RPM
+
+    def setShooterAngle(self, angle: Rotation2d) -> None:
+        self.targetAngle = min(max(Rotation2d(0), angle), constants.kShooterMaxAngle)
+
+        self.angleMotor.set(
+            Talon.ControlMode.Position,
+            self.targetAngle.degrees()
```

Comment:
nitpick but use radians :)
internally it does save some cycles since Rotation2ds are stored in radians